extensions/cli/cli.go

```diff
@@ -0,0 +1,149 @@
+package cli
+
+import (
+	ranchercli "github.com/rancher/shepherd/clients/ranchercli"
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	rancher    = "./rancher"
+	context    = "context"
+	projects   = "projects"
+	namespaces = "namespaces"
+	catalog    = "catalog"
+)
+
+// SwitchContext will display the current context and switch to the default one.
+func SwitchContext(session *session.Session) error {
```

Comment:
Why are these extensions, and not member functions of the client?

Reply:
Once you answer that it would make sense why you create new clients for each one.

Reply:
When I initially talked to Ugur on transitioning this over to clients, the initial code had `downloads.go` not in the format of a client. When we discussed, at the time it made more sense to keep the `cli.go` as extensions still and to call the `Create` and `Exists` functions.

Reply:
But you're creating new clients each time which doesn't makes sense. I would like to hear what Ugur says, but these seem that they need to be member functions for the cli client. Reinitializing the client each time is a no go

Reply:
Or at least have it take a cli client as a parameter. I will wait to see what Ugur says as well.

Reply:
Maybe we can talk offline of this. I'm still not fully knowledgeable of clients and best practices, so it would be good to get up to speed on what to do versus what not to do.

Reply:
I believe this may be resolved now? Planning to discuss offline to verify if this is the case.